src/linux/backend-v4l2.cpp

```diff
@@ -546,6 +546,65 @@ namespace librealsense
             }
         }
 
+        std::string get_devname_from_video_path(const std::string& video_path)
+        {
+            std::string dev_name;
+
+            std::ifstream uevent_file(video_path + "/uevent");
+            if (!uevent_file)
+            {
+                LOG_INFO("Cannot access " + video_path + "/uevent");
+                return dev_name;
+            }
+
+            std::string uevent_line, major_string, minor_string;
+            while (std::getline(uevent_file, uevent_line) && (major_string.empty() || minor_string.empty()))
+            {
+                if (uevent_line.find("MAJOR=") != std::string::npos)
+                {
+                    major_string = uevent_line.substr(uevent_line.find_last_of('=') + 1);
+                }
+                else if (uevent_line.find("MINOR=") != std::string::npos)
+                {
+                    minor_string = uevent_line.substr(uevent_line.find_last_of('=') + 1);
+                }
+            }
+            uevent_file.close();
+
+            if (major_string.empty() || minor_string.empty())
+            {
+                LOG_INFO("No Major or Minor number found for " + video_path);
+                return dev_name;
+            }
+
+            DIR * dir = opendir("/dev");
+            if (!dir)
+            {
+                LOG_INFO("Cannot access /dev");
+                return dev_name;
+            }
+            while (dirent * entry = readdir(dir))
+            {
+                std::string name = entry->d_name;
+                std::string path = "/dev/" + name;
+
+                struct stat st = {};
+                if (stat(path.c_str(), &st) < 0)
+                {
+                    throw linux_backend_exception(rsutils::string::from() << "Cannot identify '" << path);
```

Comment:
Shouldn't we close the dir here?

Reply:
Lets also add errno string to the error?

Reply:
I'm actually going to remove the `throw` and and just `continue` instead. It shouldn't matter to us if some random `/dev` device is inaccessible, as I'm looping through them all just to find a match with the major/minor from sysfs. If such a device exists but is inaccessible, it shouldn't be matched anyway and skipped.